packages/seacas/libraries/ioss/src/text_mesh/Iotm_TextMeshUtils.h

```diff
@@ -14,593 +14,1778 @@
+}
+
+inline void convert_to_lower_case(std::string &str)
+{
+  std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+}
+
+inline bool is_number(const std::string &str)
+{
+  for (char const &c : str) {
+    if (std::isdigit(c) == 0) return false;
+  }
+  return true;
+}
+
+template <typename T>
+std::set<T> transform_to_set(const std::vector<T> &dataAsVector)
+{
+  std::set<T> dataAsSet;
+
+  for (const T &data : dataAsVector) {
+    dataAsSet.insert(data);
+  }
+
+  return dataAsSet;
+}
+
+inline std::pair<unsigned, bool> get_id_from_part_name(const std::string &name, const std::string &prefix)
+{
+  const unsigned prefixLength = prefix.length();
+
+  if (name.length() < prefixLength + 1) return std::make_pair(0, false);
+
+  const std::string namePrefix = name.substr(0, prefixLength);
+  const std::string nameSuffix = name.substr(prefixLength);
+
+  if (strcasecmp(namePrefix.c_str(), prefix.c_str()) != 0) return std::make_pair(0, false);
+
+  unsigned id;
+  std::istringstream nameSuffixStream(nameSuffix);
+  nameSuffixStream >> id;
+  if (nameSuffixStream.fail()) {
+    return std::make_pair(0, false);
+  }
+  return std::make_pair(id, true);
+}
+
+template <typename T>
+class TopologyMapping
+{
+ public:
+  using Topology = T;
+
+  virtual ~TopologyMapping() {}
+
+  Topology topology(const std::string &textMeshName) const
+  {
+    auto it = m_nameToTopology.find(textMeshName);
+    return (it != m_nameToTopology.end() ? it->second : invalid_topology());
+  }
+
+  virtual Topology invalid_topology() const = 0;
+  virtual void initialize_topology_map() = 0;
+
+ protected:
+  std::unordered_map<std::string, Topology> m_nameToTopology;
+};
+
+class PartIdMapping
+{
+ public:
+  PartIdMapping() : m_idsAssigned(false)
+  {
+    set_error_handler([](const std::ostringstream &errmsg) { default_error_handler(errmsg); });
+  }
+
+  void register_part_name(const std::string &name)
+  {
+    m_partNames.push_back(name);
+    handle_block_part(name);
+  }
+
+  void register_part_name_with_id(const std::string &name, unsigned id)
+  {
+    register_part_name(name);
+    assign(name, id);
+  }
+
+  unsigned get(const std::string &name) const
+  {
+    if (!m_idsAssigned) assign_ids();
+    return get_part_id(name);
+  }
+
+  std::string get(unsigned id) const
+  {
+    if (!m_idsAssigned) assign_ids();
+    return get_part_name(id);
+  }
+
+  unsigned size() const
+  {
+    if (!m_idsAssigned) assign_ids();
+    return m_ids.size();
+  }
+
+  std::vector<std::string> get_part_names_sorted_by_id() const
+  {
+    if (!m_idsAssigned) assign_ids();
+
+    std::vector<std::string> names;
+    names.reserve(m_parts.size());
+
+    for (auto iter : m_parts) {
+      names.push_back(iter.second);
     }
 
-    template <typename T> class TopologyMapping
-    {
-    public:
-      using Topology = T;
+    return names;
+  }
 
-      virtual ~TopologyMapping() {}
+  std::vector<std::string> get_part_names() const { return m_partNames; }
```

Comment:
Should this be `const std::vector<std::string> &get_part_names()`  It looks like the returned vector is never modified, so a const reference would be more efficient.

Reply:
Yep, you're right

Reply:
This is fixed in the latest commit